src/reporter/pegasus_counter_reporter.cpp

```diff
@@ -167,21 +225,63 @@ void pegasus_counter_reporter::update()
 
         bool first_append = true;
         _falcon_metric.timestamp = timestamp;
+
         perf_counters::instance().iterate_snapshot(
             [&oss, &first_append, this](const dsn::perf_counters::counter_snapshot &cs) {
                 _falcon_metric.metric = cs.name;
                 _falcon_metric.value = cs.value;
                 _falcon_metric.counterType = "GAUGE";
+
                 if (!first_append)
                     oss << ",";
                 _falcon_metric.encode_json_state(oss);
                 first_append = false;
             });
         oss << "]";
-
         update_counters_to_falcon(oss.str(), timestamp);
     }
 
+    if(_enable_prometheus) {
+
+        using namespace prometheus;
+        const auto labels = Gateway::GetInstanceLabel(GetHostName());
+        Gateway gateway{_prometheus_host, std::to_string(_prometheus_port), "pegasus", labels};
+        auto registry = std::make_shared<Registry>();
+
+        perf_counters::instance().iterate_snapshot(
+            [&registry, this](const dsn::perf_counters::counter_snapshot &cs) {
+                std::string metrics_name = cs.name;
+                
+                replace(metrics_name.begin(),metrics_name.end(),'@',':');
+                replace(metrics_name.begin(),metrics_name.end(),'.','_');
+                replace(metrics_name.begin(),metrics_name.end(),'*','_');
+                replace(metrics_name.begin(),metrics_name.end(),'(','_');
+                replace(metrics_name.begin(),metrics_name.end(),')','_');
+                ddebug("%s", cs.name.c_str());
```

Comment:
if(_enable_logging){
   // print metrics names
}
We do not need print metrics when not enable logging